src/internal/cache-client.ts

```diff
@@ -413,6 +420,156 @@ export class CacheClient {
     });
   }
 
+  public async listConcatenateBack(
+    cacheName: string,
+    listName: string,
+    values: string[] | Uint8Array[],
+    ttl: CollectionTtl = CollectionTtl.fromCacheTtl(),
+    truncateFrontToSize?: number
+  ): Promise<CacheListConcatenateBack.Response> {
+    try {
+      validateCacheName(cacheName);
+      validateListName(listName);
+    } catch (err) {
+      return new CacheListConcatenateBack.Error(
+        normalizeSdkError(err as Error)
+      );
+    }
+
+    this.logger.trace(
+      `Issuing 'listConcatenateBack' request; listName: ${listName}, values length: ${
+        values.length
+      }, ${ttl.toString()}, truncateFrontToSize: ${
+        truncateFrontToSize?.toString() ?? 'null'
+      }`
+    );
+
+    const result = await this.sendListConcatenateBack(
+      cacheName,
+      this.convert(listName),
+      this.convertArray(values),
+      ttl.ttlMilliseconds() || this.defaultTtlSeconds * 1000,
+      ttl.refreshTtl(),
+      truncateFrontToSize
+    );
+    this.logger.trace(
+      `'listConcatenateBack' request result: ${result.toString()}`
+    );
+    return result;
+  }
+
+  private async sendListConcatenateBack(
+    cacheName: string,
+    listName: Uint8Array,
+    values: Uint8Array[],
+    ttlMilliseconds: number,
+    refreshTtl: boolean,
+    truncateFrontToSize?: number
+  ): Promise<CacheListConcatenateBack.Response> {
+    const request = new grpcCache._ListConcatenateBackRequest({
+      list_name: listName,
+      values: values,
+      ttl_milliseconds: ttlMilliseconds,
+      refresh_ttl: refreshTtl,
+      truncate_front_to_size: truncateFrontToSize,
+    });
+    const metadata = this.createMetadata(cacheName);
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
weird u need this lint ignore? I see you using `resolve`?  Or is it because you don't have `reject` defined?

Reply:
also why `GrpcClientWrapper` doesnt support async natively? Isnt that something we wrote? Just us doing all these Promise wrappers seems like something off? Maybe missing some context

Reply:
Just looked some more at class def. @schwern wdut? Couldn't we have a helper function wrap this in promise down in wrapper client vs make all call sites do it?  Fine if follow up PR as well.

Reply:
@eaddingtonwhite That `eslint-disable-next-line` was just cut & code from an existing method. We used to check `if(err)` to `resp` was unused. It's no longer necessary, I've removed it.

As for the rest, I just learned TypeScript last week. 🤷 I've been doing a lot of refactoring of this SDK myself. If you see something that could be done better, its likely it can be.

Anything to reduce the amount of copypasta.